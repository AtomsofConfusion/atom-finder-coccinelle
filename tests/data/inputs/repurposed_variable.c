#include <stdio.h>

int fun1(int a) {
  a = 1;
  a += 2;
  a++;
  return a;
}

int fun2(int a) {
  int b = a;
  return b;
}

int fun3(int* a) {
  *a = 1;
  return 1;
}

int fun4(int a) {
  if (a > 0) {
    a = 3;
  } 
  
  if (a > 0) {
    a = 3;
  } else if (a < 0) {
    a = -3;
  } else {
    a = 0;
  }
  
  return a;
}

int main() {
  return 0;
}