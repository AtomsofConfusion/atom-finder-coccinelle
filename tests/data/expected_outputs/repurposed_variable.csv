repurposed_variable,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/repurposed_variable.c,30,4,30,9,a = 0
repurposed_variable,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/repurposed_variable.c,4,2,4,7,a = 1
repurposed_variable,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/repurposed_variable.c,38,2,38,7,a = 1
repurposed_variable,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/repurposed_variable.c,26,4,26,9,a = 3
repurposed_variable,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/repurposed_variable.c,22,4,22,9,a = 4
repurposed_variable,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/repurposed_variable.c,28,4,28,10,a = - 3
repurposed_variable,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/repurposed_variable.c,5,2,5,8,a += 2
repurposed_variable,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/repurposed_variable.c,41,2,41,14,b = func1 ( b )
repurposed_variable,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/repurposed_variable.c,6,2,6,5,a ++
repurposed_variable,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/repurposed_variable.c,40,4,40,5,++a
repurposed_variable,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/repurposed_variable.c,39,4,39,5,--b

